File: Easy/Filter/String/1967. Number of Strings That Appear as Substrings in Word.py
"""
https://leetcode.com/problems/number-of-strings-that-appear-as-substrings-in-word/
"""


def numOfStrings(patterns: list[str], word: str) -> int:
    count = 0
    for pattern in patterns:
        if pattern in word:
            count += 1
    return count

    ## 2 method
    # return sum(x in word for x in patterns)


patterns = ["a", "a", "a"]
word = "ab"
print(numOfStrings(patterns, word))

File: PHP/Easy/Filters/String/1967. Number of Strings That Appear as Substrings in Word.php
<?php
function numOfStrings($patterns, $word)
{
    $count = 0;
    foreach ($patterns as $pattern) {
        if (strpos($word, $pattern) !== false) {
            $count += 1;
        }
    }
    return $count;
}

$patterns = ["a", "abc", "bc", "d"];
$word = "abc";
echo numOfStrings($patterns, $word);